runtime/src/iree/hal/command_buffer.c

```diff
@@ -131,6 +132,40 @@ IREE_API_EXPORT iree_string_view_t iree_hal_command_category_format(
                                      out_temp);
 }
 
+//===----------------------------------------------------------------------===//
+// iree_hal_collective_element_t
+//===----------------------------------------------------------------------===//
+
+IREE_API_EXPORT iree_status_t iree_hal_collective_element_byte_count(
```

Comment:
We try to keep all validation in the validation layer in order to ensure all backends have consistent behavior and that it can be turned off when not required. Since the validation is already checking that the enum value is in range (by testing IREE_HAL_COLLECTIVE_ELEMENT_TYPE_MAX_VALUE) this can instead be turned into a function returning the size instead of a status with an IREE_ASSERT in the default case.
```suggestion
IREE_API_EXPORT iree_device_size_t iree_hal_collective_element_byte_count(
```